server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/ScheduleService.java

```diff
@@ -441,11 +441,10 @@ public String getLog(Long scheduleId, Long taskId, OdcTaskLogLevel logLevel) {
     }
 
     public boolean hasExecutingAsyncTask(ScheduleEntity schedule) {
-        Set<Long> executingTaskIds = serviceTaskRepository.findByScheduleIdAndTaskType(
-                schedule.getId(), TaskType.ASYNC).stream()
-                .filter(entity -> !FlowNodeStatus.isFinalStatus(entity.getStatus()))
-                .map(ServiceTaskInstanceEntity::getTargetTaskId).collect(
-                        Collectors.toSet());
+        Set<Long> executingTaskIds = serviceTaskRepository.findByScheduleIdAndTaskTypeAndStatusIn(schedule.getId(),
```

Comment:
It looks like the impact on SQL plans is controllable, but it would be better to let QA test the concurrency cases.